examples/LLM/llama2/chat_app/torchserve_server_app.py

```diff
@@ -0,0 +1,200 @@
+
+
+def scale_workers(workers):
+    if st.session_state.registered:
+        num_workers = st.session_state[workers]
+        url = (
+            "http://localhost:8081/models/"
+            + MODEL_NAME
+            + "?min_worker="
+            + str(num_workers)
+            + "&synchronous=true"
+        )
+        res = requests.put(url)
+        # if res.status_code != 200:
+        #    state_container.write(res.text)
+
+        server_state_container.caption(res.text)
+
+
+def set_batch_size(batch_size):
+    if st.session_state.registered:
+        url = "http://localhost:8081/models/" + MODEL_NAME + "/1.0"
+        res = requests.delete(url)
+        server_state_container.caption(res.text)
+        st.session_state.registered = False
+
+        batch_size = st.session_state[batch_size]
+        url = (
+            "http://localhost:8081/models?model_name="
+            + MODEL_NAME
+            + "&url="
+            + MODEL_NAME
+            + ".tar.gz"
+            + "&batch_size="
+            + str(batch_size)
+            + "&initial_workers="
+            + str(workers)
+            + "&synchronous=true"
+            + "&max_batch_delay="
+            + str(max_batch_delay)
+        )
+        _register_model(url)
+
+
+def set_max_batch_delay(max_batch_delay):
+    if st.session_state.registered:
+        url = "http://localhost:8081/models/" + MODEL_NAME + "/1.0"
+        res = requests.delete(url)
+        server_state_container.caption(res.text)
+        st.session_state.registered = False
+
+        max_batch_delay = st.session_state[max_batch_delay]
+        url = (
+            "http://localhost:8081/models?model_name="
+            + MODEL_NAME
+            + "&url="
+            + MODEL_NAME
+            + ".tar.gz"
+            + "&batch_size="
+            + str(batch_size)
+            + "&initial_workers="
+            + str(workers)
+            + "&synchronous=true"
+            + "&max_batch_delay="
+            + str(max_batch_delay)
+        )
+        _register_model(url)
+
+
+if "started" not in st.session_state:
+    st.session_state.started = False
+if "stopped" not in st.session_state:
+    st.session_state.stopped = False
+if "registered" not in st.session_state:
+    st.session_state.registered = False
+
+# Replicate Credentials
+with st.sidebar:
+    st.title("🦙💬 Llama 2 TorchServe Server ")
+
+    st.button("Start Server", on_click=start_server)
+    st.button("Stop Server", on_click=stop_server)
+    st.button("Register Llama2", on_click=register_model)
+    workers = st.sidebar.slider(
+        "Num Workers",
+        key="Num Workers",
+        min_value=1,
+        max_value=4,
+        value=1,
+        step=1,
+        on_change=scale_workers,
+        args=("Num Workers",),
+    )
+    batch_size = st.sidebar.select_slider(
+        "Batch Size",
+        key="Batch Size",
+        options=[2**j for j in range(0, 8)],
+        on_change=set_batch_size,
```

Comment:
so how does batch size work here exactly? Like I'm typing things in chat does the server then respond multiple times at once?

Reply:
For each batch size, we will unregister the model and register with the new batch size. The app doesn't respond till the previous batch_size setup is completely executed. So, its pretty neat and you can't break it by moving the slider around